docs/proposals/EdgeSite.md

```diff
@@ -0,0 +1,77 @@
+---
+title: EdgeSite Design
+status: Pending
+authors: 
+  - "@cindyxing"
+approvers:
+  - "@qizha"
+  - "@kevin-wangzefeng"
+  - "@m1093782566"
+---
+# EdgeSite: Lightweight Cluster at edge
+
+## Abstract
+In Edge computing, there are scenarios where customers would like to have an whole cluster installed at edge location. As a result, 
+admins/users can leverage the local control plane to implement management functionalities and take advantages of all edge computing's benefits. 
+
+This design doc is to enable customers deploy and run lightweight clusters at edge. 
+
+## Motivation
+For these use cases, when considering large set of edge locations, the resource usage of management plane adding together will be high. 
+K3s is an opensource project enabling lightweight kubernetes cluster including the light control plane and worker components. 
+KubeEdge is the other opensource project enabling lightweight k8s client on the edge nodes. The KubeEdge agent runtime memory footprint is 
+10MB and it supports kubernetes types. Meanwhile KubEdge is targeted for Edge/IOT computing with many functionality enablements. 
+
+By integrating KubeEdge and K3s, this proposal is to allow customers to run an efficient kubernetes cluster for Edge/IOT computing. 
+
+## Assumptions
+Here we assume a cluster is deployed at edge location including the management control plane. 
+For the management control plane to manage some scale of edge worker nodes, the hosting master node needs to have sufficient resources. 
+The assumptions are
+1. EdgeSite cluster master node is of >2 CPUs and >1GB memories
+2. If high availability is required, 2-3 master nodes are needed at different edge locations
+3. The same Kubernetes security (authN and authZ) mechanisms are used to ensure the secure handshake between master and slave
+4. The same K8s HA mechanism is to be used to enable HA
+
+## Architecture Design
+<img src="../images/EdgeSite_arch.PNG"/>
```

Comment:
1. In the architecture, the other components of kubeedge on edge side are missing like devicetwin ,  metamanager , eventbus etc. Do we have a plan to integrate them as well ? 
2. Do we plan to run edgecontroller and kube-proxy as beehive modules on edge side ?
3. With kube-proxy on the edge node , will our edge service mesh design get affected ? Will we disable it ?
4. In the architecture, we intend to reuse the tunnel proxy from k3s ? What about the k3s agent -- does edged replace it ?
5. In the diagram , edged doesn't have a connection to the tunnel proxy ?